.drone.yml

```diff
@@ -12,72 +13,89 @@ steps:
       mount:
         - ./.tox
         - ./node_modules
+        - ./.pip-cache
       volumes:
         - /tmp/cache:/cache
 
   - name: setup
-    image: registry.online.ntnu.no/dotkom/onlineweb4-testbase:3.7
+    image: registry.online.ntnu.no/dotkom/onlineweb4-testbase:3.7-poetry
+    environment:
+      PIP_CACHE_DIR: ./.pip-cache
     pull: true
     when:
       event:
         - push
     commands:
       - cp onlineweb4/settings/example-local.py onlineweb4/settings/local.py
       - yarn install --pure-lockfile
-      - pip install tox
+      - poetry export -f requirements.txt > requirements.txt --dev -E prod --without-hashes
+      - pip download -r requirements.txt # Set up cache so that lint and test can run in parallell
     depends_on:
       - restore-cache
 
   - name: npm-build
-    image: registry.online.ntnu.no/dotkom/onlineweb4-testbase:3.7
+    image: registry.online.ntnu.no/dotkom/onlineweb4-testbase:3.7-poetry
     when:
       event:
         - push
-    group: testing
     commands:
       - npm run build:prod
     depends_on:
      - setup
 
   - name: js-lint
-    image: registry.online.ntnu.no/dotkom/onlineweb4-testbase:3.7
+    image: registry.online.ntnu.no/dotkom/onlineweb4-testbase:3.7-poetry
     when:
       event:
         - push
-    group: testing
     commands:
       - npm run lint-js
     depends_on:
       - setup
 
   - name: less-lint
-    image: registry.online.ntnu.no/dotkom/onlineweb4-testbase:3.7
+    image: registry.online.ntnu.no/dotkom/onlineweb4-testbase:3.7-poetry
     when:
       event:
         - push
-    group: testing
     commands:
       - npm run lint-less
     depends_on:
      - setup
 
   - name: python-lint
-    image: registry.online.ntnu.no/dotkom/onlineweb4-testbase:3.7
+    image: registry.online.ntnu.no/dotkom/onlineweb4-testbase:3.7-poetry
+    environment:
+      PIP_CACHE_DIR: ./.pip-cache
     when:
       event:
         - push
-    group: testing
     commands:
+      - sleep 30 # ensure python checks can run in parallell (they share pip-cache)
```

Comment:
this seems very error prone and dangerous

Reply:
Yes. This also didn't work because it slept across steps in Drone. I serialized the Drone build now, as a potential solution to this.